core/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -1259,28 +1260,40 @@ public static String digestToString(long digest) {
         private long writtenBytes;
         private final long checksumPosition;
         private String actualChecksum;
+        private final byte[] footerChecksum = new byte[8]; // this holds the actual footer checksum data written by to this output
 
         LuceneVerifyingIndexOutput(StoreFileMetaData metadata, IndexOutput out) {
             super(out);
             this.metadata = metadata;
-            checksumPosition = metadata.length() - 8; // the last 8 bytes are the checksum
+            checksumPosition = metadata.length() - 8; // the last 8 bytes are the checksum - we store it in footerChecksum
         }
 
         @Override
         public void verify() throws IOException {
+            String footerChecksum = null;
```

Comment:
I find it really confusing to have `footerChecksum` at the top-level that is a byte array and `footerChecksum` here that is a string, with only `this.` to indicate which one is being used.